### PR TITLE
executor: fix firstrow/max/min(bit col) error

### PR DESCRIPTION
**Pull request by XuHuaiyu**
## What have you changed? (mandatory)
Before this PR:
``` sql
CREATE TABLE `t` (
  `a` bit(1) NOT NULL,
  PRIMARY KEY (`a`)
);
insert into t value(1);
select max(a) from t group by a;
```
we'll get panic
```
lastCmd select max( a) from t group by
a, runtime error: index out of range
```

After this PR:
``` sql
tidb> select max(a) from t group by a;
+---+
| a |
+---+
|  |
+---+
1 row in set (0.00 sec)
```

## What is the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)
integration test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)
no

## Does this PR need to be added to the release notes? (mandatory)
no

## Refer to a related PR or issue link (optional)
#6952

### CODE REVIEW

File: executor/aggregate_test.go [hunk 1]
```diff
@@ -309,6 +309,13 @@ func (s *testSuite) TestAggregation(c *C) {
 	result = tk.MustQuery("select t.id, count(95), sum(95), avg(95), bit_or(95), bit_and(95), bit_or(95), max(95), min(95), group_concat(95) from t left join s on t.id = s.id")
 	result.Check(testkit.Rows("1 1 95 95.0000 95 95 95 95 95 95"))
 	tk.MustExec("set @@tidb_hash_join_concurrency=5")
+
+	// test agg bit col
+	tk.MustExec("drop table t")
+	tk.MustExec("create table t(a bit(1))")
+	tk.MustExec("insert into t value(1), (0)")
+	result = tk.MustQuery("select cast(a as signed) as idx, cast(max(a) as signed),  cast(min(a) as signed) from t group by 1 order by idx")
```
Comment (lamxTyler) [inline executor/aggregate_test.go]: It does not panic?

**Comment by XuHuaiyu** [comment]
PTAL @IamxTyler

File: executor/aggregate_test.go [hunk 2]
```diff
@@ -309,6 +309,15 @@ func (s *testSuite) TestAggregation(c *C) {
 	result = tk.MustQuery("select t.id, count(95), sum(95), avg(95), bit_or(95), bit_and(95), bit_or(95), max(95), min(95), group_concat(95) from t left join s on t.id = s.id")
 	result.Check(testkit.Rows("1 1 95 95.0000 95 95 95 95 95 95"))
 	tk.MustExec("set @@tidb_hash_join_concurrency=5")
+
+	// test agg bit col
+	tk.MustExec("drop table t")
+	tk.MustExec("CREATE TABLE `t` (`a` bit(1) NOT NULL, PRIMARY KEY (`a`))")
+	tk.MustExec("insert into t value(1), (0)")
+	tk.MustQuery("select a from t group by 1")
+	tk.MustQuery("select max(a) from t group by a")
```
Comment (lamxTyler) [inline executor/aggregate_test.go]: Check their results?

**Pull request merged**